dnsapi/dns_namecheap.sh

```diff
@@ -94,6 +110,10 @@ _get_root() {
       #not valid
       return 1
     fi
+    if ! _contains "$h" "\\."; then
```

Comment:
I added this small guard branch because $h was being set to the TLD, which obviously can not be the only part of the domain name.  However, since my domain was not being returned in the response from the `getList` api call, the `if` statement a few lines down was finding that $response *does* contain the string "com" (unsurprisingly).  It was then setting the _sub_domain and _domain variables both to "com" (obviously wrong) and returning success.

With my update, this method will correctly return false for domains which are not returned in $response.  The _get_root method will then fall back to trying another method of verifying that this user has access to the requested domain.